kie-api/src/main/java/org/kie/api/runtime/query/AdvancedQueryContext.java

```diff
@@ -0,0 +1,66 @@
+/*
+* Copyright ${year} Red Hat, Inc. and/or its affiliates.
+* 
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+* 
+*     http://www.apache.org/licenses/LICENSE-2.0
+* 
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.kie.api.runtime.query;
+
+/**
+ * 
+ * Extension class to the basic ordering and pagination queries which use <code>QueryContext</code>.  The AdvancedQueryContext is intended for use wherever advanced query functionality provided by KIE server RESTful resources is required.
```

Comment:
would be good to break the line, though the main comment is that it's not only for KIE Server but also (and maybe primary it's for the QueryService